### ISSUE TITLE
Cmake build uses deprecated FindPythonInterp and FindPythonLibs

### ISSUE THREAD

**Issue by manopapad** [StanfordLegion/legion#1568]
@Jacobfaib reported this

The following warning gets emitted with recent cmake:

```
CMake Warning (dev) at build/debug-clang/_deps/legion-src/CMakeLists.txt:282 (find_package):
  Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
  are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at build/debug-clang/_deps/legion-src/CMakeLists.txt:283 (find_package):
  Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
  are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

This warning is for project developers.  Use -Wno-dev to suppress it.
```

See https://cmake.org/cmake/help/latest/policy/CMP0148.html for context.

**Comment by elliottslaughter** [comment]
I'm happy to accept PRs for this. `FindPython3`, which is advertised as the replacement for this, is available in CMake 3.12, so we should be good to go as far as that is concerned.

**Comment by elliottslaughter** [comment]
We migrated to `FindPython3`.
